tests/test_toa_writer.py

```diff
@@ -49,6 +49,25 @@ def test_roundtrip_topo_toa_Tempo2format(self):
         assert np.abs(ts.table["mjd"][0] - ts2.table["mjd"][0]) < 1.0e-15 * u.d
         assert np.abs(ts.table["tdb"][0] - ts2.table["tdb"][0]) < 1.0e-15 * u.d
 
+    def test_commenting_toas(self):
+        # Create a barycentric TOA
+        t1time = Time(58534.0, 0.0928602471130208, format="mjd", scale="utc")
+        t1 = toa.TOA(t1time, obs="gbt", freq=0.0)
+        t2 = toa.TOA(t1time, obs="gbt", freq=0.0)
+        ts = toa.get_TOAs_list([t1, t2], ephem="DE421")
+        assert ts.ntoas == 2
+        ts.table[1]["flags"]["ignore"] = True  # flag one TOA
+        ts.write_TOA_file("testtopo.tim", format="Tempo2")
+        ts2 = toa.get_TOAs("testtopo.tim")
+        assert ts2.ntoas == 2  # none should be commented
+        ts.write_TOA_file("testtopo.tim", format="Tempo2", commentflag=None)
+        ts3 = toa.get_TOAs("testtopo.tim")
+        assert ts3.ntoas == 2  # none should be commented
+        ts.write_TOA_file("testtopo.tim", format="Tempo2", commentflag="ignore")
+        ts4 = toa.get_TOAs("testtopo.tim")
+        assert ts4.ntoas == 1  # one should be commented
+        print(ts.table, ts2.table, ts3.table, ts4.table)
```

Comment:
This will never do anything - `pytest` only prints the output if an exception arose, in which case this line is never executed.

Reply:
Those print() functions were in each of the class methods from before.  I was just carrying them along.  (and note that they *are* executed and actually print stuff if you use the `-s` option for pytest (which I use a lot for debugging tests).